index.js

```diff
@@ -169,7 +169,7 @@ function markdownTable(table, options) {
 
     index = -1;
 
-    while (++index < cellCount) {
+    while (settings.straighten !== false && ++index < cellCount) {
```

Comment:
That was my original approach but the only problem when doing this is that I received a warning from the linter about too many nested blocks.

I thought about re-factoring the code quite heavily to move some of the logic out into self contained functions, but didn't want to re-factor it too much without knowing what you thought.